src/main/java/org/apache/commons/imaging/formats/png/PngWriter.java

```diff
@@ -596,8 +588,10 @@ public void writeImage(final BufferedImage src, final OutputStream os, Map<Strin
             // Debug.debug("uncompressed", uncompressed.length);
 
             final ByteArrayOutputStream baos = new ByteArrayOutputStream();
-            final DeflaterOutputStream dos = new DeflaterOutputStream(baos);
             final int chunkSize = 256 * 1024;
+            Deflater deflater = new Deflater(Deflater.BEST_COMPRESSION);
```

Comment:
Hi @adorn what if we keep the default behaviour (i.e. use default compression level) but allow users to change it?

See https://github.com/adorn/commons-imaging/pull/1

Reply:
That's also OK.  Should I add some PngConstants?

PARAM_KEY_PNG_DEFLATE_COMPRESSION_LEVEL  ="PNG_DEFLATE_COMPRESSION_LEVEL";

    public static final int PNG_DEFLATE_COMPRESSION_LEVEL_DEFAULT= Deflater.DEFAULT_COMPRESSION;
    public static final int PNG_DEFLATE_COMPRESSION_LEVEL_BEST_COMPRESSION= Deflater.BEST_COMPRESSION;
    public static final int PNG_DEFLATE_COMPRESSION_LEVEL_BEST_SPEED= Deflater.BEST_SPEED;

Reply:
I think it's not necessary. These values are already at the `Deflater` class, and may change. So we should be better allowing users to pass anything. There is also a ticket I think to allow for other compression algorithms (even though specification uses only Deflater, it says vendors can use other algorithms), in which case it could lead to having more values for each algorithm.